### PR TITLE
[5615][next] Remove Sandbox Branch from the site creation dialog

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/5615

### CODE REVIEW

File: ui/app/src/components/CreateSiteDialog/BlueprintForm.tsx [hunk 1]
```diff
@@ -91,9 +93,9 @@ const messages = defineMessages({
     id: 'createSiteDialog.cantStart',
     defaultMessage: 'Project names may not start with zeros, dashes (-) or underscores (_).'
   },
-  sandboxBranch: {
-    id: 'createSiteDialog.sandboxBranch',
-    defaultMessage: 'Sandbox Branch'
+  gitBranch: {
+    id: 'createSiteDialog.gitBranch',
+    defaultMessage: 'Git Branch'
```
Comment (rart) [inline ui/app/src/components/CreateSiteDialog/BlueprintForm.tsx:87]: There's a couple of message definitions with the same string. Please consolidate and clean up.

File: ui/app/src/components/CreateSiteDialog/BlueprintForm.tsx [hunk 2]
```diff
@@ -301,26 +286,18 @@ function BlueprintForm(props: BlueprintFormProps) {
...
-              }
-              label={formatMessage(messages.createAsOrphan)}
+            <TextField
+              id="sandboxBranch"
+              name="gitBranch"
+              label={formatMessage(messages.gitBranch)}
+              fullWidth
+              onKeyPress={onKeyPress}
+              onChange={(event) => handleInputChange(event)}
+              placeholder={'master'}
```
Comment (rart) [inline ui/app/src/components/CreateSiteDialog/BlueprintForm.tsx]:
```suggestion
              placeholder="master"
```

File: ui/app/src/components/CreateSiteDialog/BlueprintForm.tsx [hunk 3]
```diff
@@ -301,26 +286,18 @@ function BlueprintForm(props: BlueprintFormProps) {
             helperText={formatMessage(messages.descriptionMaxLength, { maxLength: maxLength })}
           />
         </Grid>
-        {blueprint.id === 'GIT' && (
+        {blueprint.id !== 'GIT' && (
```
Comment (rart) [inline ui/app/src/components/CreateSiteDialog/BlueprintForm.tsx]: Remove conditional. Remove from GitForm. Always shown, always in same position.

File: ui/app/src/components/CreateSiteDialog/BlueprintForm.tsx [hunk 4]
```diff
@@ -340,6 +317,28 @@ function BlueprintForm(props: BlueprintFormProps) {
...
+            />
+            <Typography
+              variant="subtitle2"
+              component="small"
+              className={`${classes.helpText} ${inputs.createAsOrphan ? '' : classes.muted}`}
+            >
+              {formatMessage(messages.createAsOrphanHelpText)}
+            </Typography>
+          </Grid>
+        )}
```
Comment (rart) [inline ui/app/src/components/CreateSiteDialog/BlueprintForm.tsx]: Move to GitForm. Check that the layout was fixed (helper text below)

File: ui/app/src/components/CreateSiteDialog/CreateSiteDialog.tsx [hunk 5]
```diff
@@ -625,7 +626,7 @@ function CreateSiteDialog(props: CreateSiteDialogProps) {
         patch: site.blueprint.version.patch
       }
     };
-    if (site.sandboxBranch) params.sandboxBranch = site.sandboxBranch;
+    if (site.gitBranch) params.sandboxBranch = site.gitBranch as string;
```
Comment (rart) [inline ui/app/src/components/CreateSiteDialog/CreateSiteDialog.tsx:629]: Why is it required to cast `gitBranch`? Is the type not well defined?
Also, if this is not required, does the API accept blank/missing for sandboxBranch and/or remoteBranch? Don't we need to default it?

**Review by rart — APPROVED** [review]

**Review by rart — CHANGES_REQUESTED** [review]
Pressed approve instead of request: see changes above.

File: ui/app/src/components/CreateSiteDialog/CreateSiteDialog.tsx (hunk 5 above)
Comment (jvega190) [reply ui/app/src/components/CreateSiteDialog/CreateSiteDialog.tsx:629]: When the fields are not sent (sandbox and remote) the service uses `master` as default.

**Review by rart — APPROVED** [review]

**Pull request merged**